src/index.js

```diff
@@ -172,21 +145,21 @@ const childSeedFromSeed = async config => {
 
   const childSeed = await hash(seed, salt, defaultTo(password, ''));
 
-  return childSeed.slice(0, seed.length || seed.byteLength);
+  return childSeed.slice(0, seed.byteLength || seed.length);
```

Comment:
The fact that both of these get hit now (evidenced in test coverage) is weird. Buffers should always have a `byteLength` property as far as I'm aware, so the docstrings for this function may not be correct.

As far as `hash()` is concerned passing a string is fine, but may or may not have different output?

Reply:
The docstrings say this function accepts a `Buffer`. Calling this function with a `string` instead works. This seems confusing, and potentially dangerous if any function in the chain interprets strings and Buffers as different data.